File: 0090-subsets-ii/0090-subsets-ii.cpp
class Solution
{
    public:

        void helper(int strt, vector<int> &nums, vector<int> &subset, vector< vector< int>> &subsets)
        {
            subsets.push_back(subset);
            for (int i = strt; i < nums.size(); i++)
            {
                if(strt!=i&&nums[i]==nums[i-1])continue;
                subset.push_back(nums[i]);
                helper(i + 1, nums, subset, subsets);
                subset.pop_back();
            }
        }

    vector<vector < int>> subsetsWithDup(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());
        vector<vector < int>> subsets;
        vector<int> subset;
        helper(0, nums, subset, subsets);
        return subsets;
    }
};

File: 0090-subsets-ii/0090-subsets-ii.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        int combination = (1<<(nums.length));
        List<List<Integer>> res = new ArrayList<>();
        Set<List<Integer>> st = new HashSet<>();
        System.out.println(combination);
        for(int i=0;i<combination;i++)
        {
            List<Integer> intermediate = new ArrayList<>();
            for(int j=0;j<nums.length;j++)
            {
                if((i&(1<<j))>0)
                {
                    intermediate.add(nums[j]);
                }
            }
            Collections.sort(intermediate);
            if(!st.contains(intermediate))
            {
                res.add(intermediate);
                st.add(intermediate);
            }
        }
        return res;
    }
}